srtcore/api.cpp

```diff
@@ -851,6 +850,8 @@ int CUDTUnited::connect(const SRTSOCKET u, const sockaddr* name, int namelen, in
       throw e;
    }
 
+   s->m_Status = SRTS_CONNECTING;
```

Comment:
Why not simply move this instruction to inside `startConnect` to make it happen after setting `m_bConnecting = true`?